### PR TITLE
docs: Deprecate GO

### PR DESCRIPTION
**Pull request by aleksander-dytko**
## Description
Inform users about GO Client deprecation and move to the community hub. This will be supported with a [blogpost communication](https://docs.google.com/document/d/12i2yQSPnjOB4l1qxO5-no9qGDWlHc3BKvyc5WdD97tU/edit).

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: clients/go/README.md [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+
+### What Does This Mean for Users?
+
+* No New Features or Updates: Starting with Camunda 8.5-alpha1, the Go Client will no longer receive new features, updates, or official support from Camunda.
+* Community Maintenance: The Go Client will be moved to [Camunda Community Hub](https://github.com/camunda-community-hub) and can be maintained by the community. We encourage contributions from users who wish to continue using and improving this client.
+* Transition to REST API: We recommend users transition to the Camunda 8 REST API for all future development. The REST API provides comprehensive functionality and is supported by tools such as cURL, Postman, and OpenAPI.
+
+### Future Considerations
+We value feedback from our community. Based on user input, we may explore developing a new client for the Camunda 8 REST API, based on a different technology that aligns with our strategic goals and internal expertise.
+For more information on the deprecation and our API strategy, please refer to the official [Camunda blog](https://camunda.com/blog/).
```
Comment (akeller) [inline clients/go/README.md:18]: Will there be a specific blog on this API topic?

**Review by megglos — APPROVED** [review]
small suggestion on highlighting that old minor releases are still supported

**Comment by megglos** [comment]
@aleksander-dytko I rebased and fixed the commit message to adhere to our linting

**Comment by akeller** [comment]
I would recommend a tech writer review this content. Please reach out on ask-c8-documentation if you want a retro review.

**Comment by aleksander-dytko** [comment]
@camunda/tech-writers could you help with a retro-review of this PR?

**Comment by mesellings** [comment]
@aleksander-dytko I will review this for you 👍

File: clients/go/README.md [hunk 2]
```diff
@@ -1,6 +1,25 @@
 # Zeebe Go Client
 
-## Development
+> [!WARNING]
+> The Zeebe Go Client will be officially deprecated with 8.6 release as part of our efforts to streamline the Camunda 8 API experience. This client will not get released with Camunda 8.6, thus no longer receive new features and will be transitioned to a community-maintained status.
```
Comment (mesellings) [inline clients/go/README.md:4]: Suggestion to align with the announcement:

The Zeebe Go Client will be officially deprecated with the 8.6 release as part of our efforts to streamline the Camunda 8 API experience. This client will not get released starting with Camunda 8.6, will no longer receive new features, and will be transitioned to a community-maintained status.

Comment (mesellings) [reply clients/go/README.md:4]: Also, do we need to mention zbctl here as well? The announcement does, e.g.:

The Zeebe Go Client and zbctl will be officially deprecated with the 8.6 release as part of our efforts to streamline the Camunda 8 API experience. This client and CLI utility will not get released starting with Camunda 8.6, will no longer receive new features, and will be transitioned to a community-maintained status.

**Review by mesellings — COMMENTED** [review]
@aleksander-dytko LGTM, just the initial warning admonition that I have left some feedback on 👍

File: clients/go/README.md (hunk 2 above)
Comment (akeller) [reply clients/go/README.md:4]: Did this get incorporated into a PR @aleksander-dytko @mesellings?

Comment (mesellings) [reply clients/go/README.md:4]: > Did this get incorporated into a PR @aleksander-dytko @mesellings?

@akeller the actual announcement was merged for the alpha 5 release as part of the PR: https://github.com/camunda/camunda-docs/pull/4264, but I'm not sure about this change to the readme file @aleksander-dytko?

**Pull request merged**